### ISSUE TITLE
feature & data shape

### ISSUE THREAD

**Issue by mhyeonsoo** [microsoft/UniVL#37]
Hi, I am leaving another question here.

I am looking at the data format for setting my custom dataset in the same format as YouCookii.
When I loaded the pickle files which are **'youcookii_data.no_transcript.pickle'** and **'youcookii_videos_features.pickle'**,
I could see the lengths of the two files are different.
I expected to see the same length for both data and feature pickles, but can you tell me why those lengths are different?

```python
data = pickle.load(open('youcookii_data.no_transcript.pickle','rb'))
len(data)
# 1790
```
```python
features = pickle.load(open('youcookii_videos_features.pickle','rb'))
len(features)
# 1905
```

Thank you so much,

**Comment by ArrowLuo** [comment]
Hi @mhyeonsoo, it is because the labels of testing videos in YouCookii are unpublished. We separately generated the pickle and did not pair them in advance,  so the data is less than the features.

**Comment by mhyeonsoo** [comment]
Thanks for the quick response @ArrowLuo, it helped a lot.

I have one more related question, which may be opened with another issue, but I am just leaving here.
When I followed instructions, to my best knowledge, it seemed like I have to make a .npy feature for the custom video to make the CSV format features.

Then I saw https://github.com/antoine77340/S3D_HowTo100M, and it shows how to extract text embedding and video embedding which has batch x 512 (or 1024) dimension.

Should I just save this batch x 512 feature as a .npy file for the rest of the generation process? Or should I follow something more than this?

Thanks,

**Comment by ArrowLuo** [comment]
Hi @mhyeonsoo, this project just needs to save batch x 512 feature to .npy. We do not need to embed the text because  the text encoder can deal with it from raw text. The format of .csv, .pickle, and .npy and be found in [readme](https://github.com/microsoft/UniVL/tree/main/dataloaders#1-s3d-features-pretrained-on-howto100m) and its sub-links, e.g., [here](https://github.com/ArrowLuo/VideoFeatureExtractor). Best~

**Comment by mhyeonsoo** [comment]
Thanks @ArrowLuo,

So now I know that I need to save batch x 512 feature using https://github.com/antoine77340/S3D_HowTo100M, 
and merge npy files into csv, and convert them into .pickle file.

And this would be the last question.
When I see a single example feature from youcookii feature file (youcookii_videos_features.pickle), 
the shape of **features['vUhoMXc7FJM']** is (1107, 1024)

Should I understand the meaning of '1107' as the number of frames in 'vUhoMXc7FJM' video?
just wanna match the dimension before I extract the feature from the videos.

Thank you so much for giving me advice.
Best regards,

**Comment by ArrowLuo** [comment]
Hi @mhyeonsoo, we have a version of extractor [https://github.com/ArrowLuo/VideoFeatureExtractor](https://github.com/ArrowLuo/VideoFeatureExtractor). Besides, the .csv, .pickle, and .npy are separate files and do not merge them together. See [https://github.com/microsoft/UniVL/tree/main/dataloaders#1-s3d-features-pretrained-on-howto100m](https://github.com/microsoft/UniVL/tree/main/dataloaders#1-s3d-features-pretrained-on-howto100m). Yes, the '1107' is the number of frames of 'vUhoMXc7FJM' video.

**Comment by mhyeonsoo** [comment]
I appreciate it @ArrowLuo :)
I will try with the reference you offered. Hopefully, I can get the custom-trained model at the end.
I will close this issue. Thanks!
